<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="scp.targets.vertx.clustered.RequesterResponderTest" tests="8" skipped="0" failures="1" errors="0" timestamp="2015-12-04T18:12:41" hostname="Matthews-MacBook-Pro.local" time="0.707">
  <properties/>
  <testcase name="testExcessLoad" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.625">
    <failure message="java.lang.AssertionError: expected:&lt;EXCESS_LOAD&gt; but was:&lt;LOCAL_TIME_OUT&gt;" type="java.lang.AssertionError">java.lang.AssertionError: expected:&lt;EXCESS_LOAD&gt; but was:&lt;LOCAL_TIME_OUT&gt;
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at scp.api.RequesterResponderTest.testExcessLoad(RequesterResponderTest.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:86)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:49)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:69)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:48)
	at sun.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:105)
	at sun.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.messaging.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.messaging.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:360)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
</failure>
  </testcase>
  <testcase name="testRemoteUnknownFailure" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.029"/>
  <testcase name="testLocalTimeOut" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.003"/>
  <testcase name="testStaleData" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.011"/>
  <testcase name="testDataUnavailable" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.013"/>
  <testcase name="testSuccessfulRequest" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.004"/>
  <testcase name="testFastRequest" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.005"/>
  <testcase name="testRemoteTimeOut" classname="scp.targets.vertx.clustered.RequesterResponderTest" time="0.016"/>
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[Dec 04, 2015 12:12:38 PM com.hazelcast.config.AbstractXmlConfigHelper
WARNING: Name of the hazelcast schema location incorrect using default
Dec 04, 2015 12:12:38 PM com.hazelcast.instance.DefaultAddressPicker
INFO: [LOCAL] [dev] [3.5.2] Prefer IPv4 stack is true.
Dec 04, 2015 12:12:38 PM com.hazelcast.instance.DefaultAddressPicker
INFO: [LOCAL] [dev] [3.5.2] Picked Address[192.168.1.2]:5704, using socket ServerSocket[addr=/0:0:0:0:0:0:0:0,localport=5704], bind any local is true
Dec 04, 2015 12:12:38 PM com.hazelcast.spi.OperationService
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Backpressure is disabled
Dec 04, 2015 12:12:38 PM com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Starting with 2 generic operation threads and 4 partition operation threads.
Dec 04, 2015 12:12:38 PM com.hazelcast.system
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Hazelcast 3.5.2 (20150826 - ba8dbba) starting at Address[192.168.1.2]:5704
Dec 04, 2015 12:12:38 PM com.hazelcast.system
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
Dec 04, 2015 12:12:38 PM com.hazelcast.instance.Node
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Creating MulticastJoiner
Dec 04, 2015 12:12:38 PM com.hazelcast.core.LifecycleService
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Address[192.168.1.2]:5704 is STARTING
Dec 04, 2015 12:12:38 PM com.hazelcast.cluster.impl.MulticastJoiner
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Trying to join to discovered node: Address[192.168.1.2]:5701
Dec 04, 2015 12:12:38 PM com.hazelcast.nio.tcp.SocketConnector
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Connecting to /192.168.1.2:5701, timeout: 0, bind-any: true
Dec 04, 2015 12:12:38 PM com.hazelcast.nio.tcp.SocketAcceptor
INFO: [192.168.1.2]:5701 [dev] [3.5.2] Accepting socket connection from /192.168.1.2:52098
Dec 04, 2015 12:12:38 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Established socket connection between /192.168.1.2:52098
Dec 04, 2015 12:12:38 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5701 [dev] [3.5.2] Established socket connection between /192.168.1.2:5701
Dec 04, 2015 12:12:39 PM com.hazelcast.cluster.ClusterService
INFO: [192.168.1.2]:5701 [dev] [3.5.2] 

Members [4] {
	Member [192.168.1.2]:5701 this
	Member [192.168.1.2]:5702
	Member [192.168.1.2]:5703
	Member [192.168.1.2]:5704
}

Dec 04, 2015 12:12:39 PM com.hazelcast.cluster.ClusterService
INFO: [192.168.1.2]:5702 [dev] [3.5.2] 

Members [4] {
	Member [192.168.1.2]:5701
	Member [192.168.1.2]:5702 this
	Member [192.168.1.2]:5703
	Member [192.168.1.2]:5704
}

Dec 04, 2015 12:12:39 PM com.hazelcast.cluster.ClusterService
INFO: [192.168.1.2]:5703 [dev] [3.5.2] 

Members [4] {
	Member [192.168.1.2]:5701
	Member [192.168.1.2]:5702
	Member [192.168.1.2]:5703 this
	Member [192.168.1.2]:5704
}

Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.SocketConnector
INFO: [192.168.1.2]:5702 [dev] [3.5.2] Connecting to /192.168.1.2:5704, timeout: 0, bind-any: true
Dec 04, 2015 12:12:39 PM com.hazelcast.cluster.ClusterService
INFO: [192.168.1.2]:5704 [dev] [3.5.2] 

Members [4] {
	Member [192.168.1.2]:5701
	Member [192.168.1.2]:5702
	Member [192.168.1.2]:5703
	Member [192.168.1.2]:5704 this
}

Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.SocketConnector
INFO: [192.168.1.2]:5703 [dev] [3.5.2] Connecting to /192.168.1.2:5704, timeout: 0, bind-any: true
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5702 [dev] [3.5.2] Established socket connection between /192.168.1.2:52099
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5703 [dev] [3.5.2] Established socket connection between /192.168.1.2:52100
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.SocketAcceptor
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Accepting socket connection from /192.168.1.2:52099
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.SocketAcceptor
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Accepting socket connection from /192.168.1.2:52100
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Established socket connection between /192.168.1.2:5704
Dec 04, 2015 12:12:39 PM com.hazelcast.nio.tcp.TcpIpConnectionManager
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Established socket connection between /192.168.1.2:5704
Dec 04, 2015 12:12:39 PM com.hazelcast.partition.InternalPartitionService
INFO: [192.168.1.2]:5701 [dev] [3.5.2] Re-partitioning cluster data... Migration queue size: 67
Dec 04, 2015 12:12:41 PM com.hazelcast.partition.InternalPartitionService
INFO: [192.168.1.2]:5701 [dev] [3.5.2] All migration tasks have been completed, queues are empty.
Dec 04, 2015 12:12:41 PM com.hazelcast.core.LifecycleService
INFO: [192.168.1.2]:5704 [dev] [3.5.2] Address[192.168.1.2]:5704 is STARTED
[vert.x-eventloop-thread-0] INFO scp.targets.vertx.CommunicationManagerImpl - Vertx creation succeeded
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[vert.x-eventloop-thread-5] ERROR scp.targets.vertx.CommunicationManagerImpl - Deserialization failed
java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2328)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:2797)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:802)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:299)
	at scp.targets.vertx.CommunicationManagerImpl.getData(CommunicationManagerImpl.java:260)
	at scp.targets.vertx.CommunicationManagerImpl.lambda$registerExecutor$21(CommunicationManagerImpl.java:226)
	at scp.targets.vertx.CommunicationManagerImpl$$Lambda$36/1603186955.handle(Unknown Source)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handleMessage(EventBusImpl.java:1121)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handle(EventBusImpl.java:1115)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$doReceive$179(EventBusImpl.java:755)
	at io.vertx.core.eventbus.impl.EventBusImpl$$Lambda$56/1361221987.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$16(ContextImpl.java:335)
	at io.vertx.core.impl.ContextImpl$$Lambda$20/1411535830.run(Unknown Source)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:358)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:357)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Thread.java:745)
Dec 04, 2015 12:12:41 PM io.vertx.core.impl.ContextImpl
SEVERE: Unhandled exception
java.lang.RuntimeException: java.io.EOFException
	at scp.targets.vertx.CommunicationManagerImpl.getData(CommunicationManagerImpl.java:266)
	at scp.targets.vertx.CommunicationManagerImpl.lambda$registerExecutor$21(CommunicationManagerImpl.java:226)
	at scp.targets.vertx.CommunicationManagerImpl$$Lambda$36/1603186955.handle(Unknown Source)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handleMessage(EventBusImpl.java:1121)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handle(EventBusImpl.java:1115)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$doReceive$179(EventBusImpl.java:755)
	at io.vertx.core.eventbus.impl.EventBusImpl$$Lambda$56/1361221987.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$16(ContextImpl.java:335)
	at io.vertx.core.impl.ContextImpl$$Lambda$20/1411535830.run(Unknown Source)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:358)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:357)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2328)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:2797)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:802)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:299)
	at scp.targets.vertx.CommunicationManagerImpl.getData(CommunicationManagerImpl.java:260)
	... 12 more

[Test worker] WARN scp.impl.RequestRequester - Timed out handling 1-th request
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
Dec 04, 2015 12:12:42 PM io.vertx.core.eventbus.impl.EventBusImpl
WARNING: Message reply handler timed out as no reply was received - it will be removed
[vert.x-eventloop-thread-5] ERROR scp.targets.vertx.CommunicationManagerImpl - Vertx Send failed: TIMEOUT
[vert.x-eventloop-thread-7] ERROR scp.impl.ResponderInvoker - Failed to handle 1-th reception
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Don't Panic! This exception is good :)
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
	at scp.impl.ResponderInvoker.invokeHandler(ResponderInvoker.java:100)
	at scp.impl.ResponderInvoker.serviceRequest(ResponderInvoker.java:87)
	at scp.targets.vertx.CommunicationManagerImpl.lambda$registerResponder$17(CommunicationManagerImpl.java:163)
	at scp.targets.vertx.CommunicationManagerImpl$$Lambda$39/804598093.handle(Unknown Source)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handleMessage(EventBusImpl.java:1121)
	at io.vertx.core.eventbus.impl.EventBusImpl$HandlerRegistration.handle(EventBusImpl.java:1115)
	at io.vertx.core.eventbus.impl.EventBusImpl.lambda$doReceive$179(EventBusImpl.java:755)
	at io.vertx.core.eventbus.impl.EventBusImpl$$Lambda$56/1361221987.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$16(ContextImpl.java:335)
	at io.vertx.core.impl.ContextImpl$$Lambda$20/1411535830.run(Unknown Source)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:358)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:357)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: Don't Panic! This exception is good :)
	at scp.api.RequesterResponderTest$3.respond(RequesterResponderTest.java:163)
	at scp.impl.ResponderInvoker$$Lambda$8/548231842.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[Test worker] WARN scp.impl.RequestRequester - Timed out handling 1-th request
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[Test worker] INFO scp.impl.RequestRequester - Handling stale 1-th response
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[Test worker] INFO scp.impl.RequestRequester - Handling stale 1-th response
[Test worker] WARN scp.impl.RequestRequester - Dropping 2-th request
[Test worker] INFO scp.targets.vertx.CommunicationManagerImpl - Done setting up vertx eventbus
[vert.x-eventloop-thread-7] WARN scp.impl.ResponderInvoker - Timed out handling 1-th reception
]]></system-err>
</testsuite>
